examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -309,37 +370,30 @@ on_event(struct pt_regs* ctx) {
     // Get raw native user stack
     struct pt_regs user_regs;
 
-    // ebpf doesn't allow direct access to ctx->cs, so we need to copy it
-    int cs;
-    bpf_probe_read_kernel(&cs, sizeof(cs), &(ctx->cs));
-
     // Are we in user mode?
-    if (cs & 3) {
+    if (user_mode(ctx)) {
       // Yes - use the registers context given to the BPF program
-      user_regs = *ctx;
+      bpf_probe_read_kernel(&user_regs, sizeof(user_regs), ctx);
```

Comment:
TODO figure this out - why these 2 lines do not have the same effect, as if `bpf_probe_read_kernel` fails.

TODO maybe because `bpf_probe_read_kernel` doesn't exist in pre-5.4, so this read fails???